lib/matplotlib/backends/_backend_tk.py

```diff
@@ -1004,29 +1004,28 @@ def new_figure_manager_given_figure(cls, num, figure):
         """
         Create a new figure manager instance for the given figure.
         """
-        with _restore_foreground_window_at_end():
-            window = Tk.Tk(className="matplotlib")
-            window.withdraw()
-
-            # Put a mpl icon on the window rather than the default tk icon.
-            # Tkinter doesn't allow colour icons on linux systems, but tk>=8.5 has
-            # a iconphoto command which we call directly. Source:
-            # http://mail.python.org/pipermail/tkinter-discuss/2006-November/000954.html
-            icon_fname = os.path.join(
-                rcParams['datapath'], 'images', 'matplotlib.ppm')
-            icon_img = Tk.PhotoImage(file=icon_fname)
-            try:
-                window.tk.call('wm', 'iconphoto', window._w, icon_img)
-            except Exception as exc:
-                # log the failure (due e.g. to Tk version), but carry on
-                _log.info('Could not load matplotlib icon: %s', exc)
-
-            canvas = cls.FigureCanvas(figure, master=window)
-            manager = cls.FigureManager(canvas, num, window)
-            if matplotlib.is_interactive():
-                manager.show()
-                canvas.draw_idle()
-            return manager
+        window = Tk.Tk(className="matplotlib")
```

Comment:
The rebase incorrectly got rid of the contextmanager.
Anyone can dismiss this review once fixed.